### PR TITLE
WIP: Create config

### PR DESCRIPTION
**Pull request by johan-bjareholt**
Need to think a few more things through to make it work in a nice manner

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ appdirs = "0.2.0"
 lazy_static = "1.2"
 log = "0.4"
 fern = { version = "0.5", features = ["colored"] }
+toml = "0.4"
```
Comment (ErikBjare) [inline Cargo.toml]: Yes! :heart:

File: src/config.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+pub fn get_config() -> AWConfig {
+    let mut config_path = dirs::get_config_dir().unwrap();
+    match Environment::active().expect("Failed to get current environment") {
+        Environment::Production => config_path.push("config.toml"),
+        Environment::Development => config_path.push("config-testing.toml"),
+        Environment::Staging => panic!("Staging environment not supported"),
+    };
+
+    if !config_path.is_file() {
+        debug!("Writing default config at {:?}", config_path);
```
Comment (ErikBjare) [inline src/config.rs]: I don't think we should write a default configuration. The defaults should be loaded in memory, optionally overloaded by whatever is specified in the config file.

I know this is not how we do it in the Python code, but I've been thinking a while that we've been doing it wrong.

Comment (johan-bjareholt) [reply src/config.rs]: Good point.
On the other hand it is nice to have all options visible in the config.
Not sure what the best option would be to be honest, both has its flaws.

Comment (ErikBjare) [reply src/config.rs]: True, although the options should be documented anyway. 

Perhaps we could add the default options to the config, but commented out?

Comment (johan-bjareholt) [reply src/config.rs]: Fixed

File: src/config.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
+use std::fs::File;
+use std::io::{Read, Write};
+use rocket::config::{Config, Environment, Limits};
+
+use dirs;
+
+#[derive(Serialize,Deserialize)]
+pub struct AWConfig {
+    #[serde(default = "default_address")]
+    address: String,
```
Comment (johan-bjareholt) [inline src/config.rs]: Maybe we should have a section for address and port though? What should we call it?

Comment (ErikBjare) [reply src/config.rs]: I don't think it's needed tbh, works fine in the top-level.

**Pull request merged**